### DISCUSSION TITLE
confusion around 'edit'

### DISCUSSION THREAD

**Discussion by laur89** [hluk/CopyQ#2080] [Q&A] [answered]
Have some questions/confusion around how edit/F2/edit() should behave.

1. Given F2 is triggered by the default command, and also an `editor` has been defined in our config, what should `F2` trigger - built-in editor or our command? Currently, for me anyway, F2 always triggers built-in editor.
1. The behavior with `ctrl+e` (ie editor) is extremely flaky - most of the time the editor doesn't contain the selected text; even if it's only `text/plain`; sometimes it does. Did enable debug, but can't see any difference between successful & non-successful times. Any pointers on how to debug this? Note editor does open file like `/tmp/CopyQ.<hash>.txt`, but it's empty and after saving it it remains on the file-system, unlike in the successful cases.
1. if our own edit command is defined (from [here](https://github.com/hluk/CopyQ/issues/1511#issuecomment-864773614)), then `F2` toggles editor command when defined, as explained in the thread. Is there a way to make it behave in default manner (as explained in the first point above; basically to keep F2 toggling internal editor, and invoking our defined editor only by `ctrl+e`)?

**Comment by hluk** [comment, marked as answer]
ad 1. F2 always opens internal editor; Ctrl+E opens external editor if configured, otherwise it uses the internal one.

ad 2. The external editor must be configured to be started in foreground within the initial process. If you launch some editors from terminal, they start a new process and the initial one exists so that user can use the terminal session while the editor is running. See: https://copyq.readthedocs.io/en/latest/faq.html#why-does-my-external-editor-fail-to-edit-items

ad 3. I don't think it is possible unless you avoid setting an external editor command.
